src/context/videosContext.jsx

```diff
@@ -0,0 +1,37 @@
+import { createContext, useContext, useReducer } from "react"
+import { VideosReducer } from "reducers/videosReducer"
+import { useAsyncServerCall } from "hooks/useAsyncServerCall"
+import { GET_CATEGORIES, GET_VIDEOS } from "utils/constants/apiEndPoints"
+
+const videosContext = createContext(null);
+
+function VideosProvider({ children }){
+
+    const [data, videosDispatch] = useReducer(VideosReducer,{
+        videos: [],
+        categories: [],
+        categoryFilter: 'All',
+    })
+    
+    function getFilteredData(category, videos){
+        if(category==='All'){
+            return [...videos]
+        }
+        console.log(videos.filter(video => video.category===category))

```

Comment:
`console.log` can be removed if not needed.

Reply:
Removed.